bin/internal/shared.bat

```diff
@@ -52,18 +52,23 @@ GOTO :after_subroutine
     CALL "%bootstrap_path%"
   )
 
-  REM Check that git exists and get the revision
-  SET git_exists=false
+  REM Check that git exists and get the revision.
+  WHERE git >NUL 2>&1
+  IF "%ERRORLEVEL%" NEQ "0" (
+    REM Could not find git. Exit without /B to avoid retrying.
+    ECHO Error: Unable to find git in your PATH. && EXIT 1
+  )
   2>NUL (
-    PUSHD "%flutter_root%"
-    FOR /f %%r IN ('git rev-parse HEAD') DO (
-      SET git_exists=true
+    REM 'FOR /f' spawns a new terminal instance to run the command, so we need
+    REM to 'PUSHD %FLUTTER_ROOT%' before getting the git revision. If an
+    REM 'AutoRun' command is defined in the user's registry, we might not be in
+    REM the directory we expect to be in.
```

Comment:
Would the following be easier to follow?

```suggestion
    REM 'FOR /f' spawns a new terminal instance to run the command. If an
    REM 'AutoRun' command is defined in the user's registry, that command could
    REM change the working directory, and then we wouldn't be in the directory
    REM we expect to be in. To prevent this, we need to 'PUSHD %FLUTTER_ROOT%'
    REM before getting the git revision.
```

It fully explains the "why" before explaining the "what."

Reply:
Yep, that's a good idea!